### PR TITLE
Convert to Circle2

### PR DESCRIPTION
**Pull request by ellisjoe**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# This file was generated by the excavator check 'excavator/manage-circleci' as specified in .circleci/template.sh.
+# To request a modification to the general template, file an issue on Excavator.
+# To manually manage the CircleCI configuration for this project, remove the .circleci/template.sh file.
+
+version: 2
+jobs:
+  build:
+    docker:
+      - image: ellisjoe/openjdk-java-8-openssl:0.0.2
```
Comment (iamdanfox) [inline .circleci/config.yml]: would prefer a repo for this to exist so it's easy to see the source / make tweaks (e.g. https://github.com/iamdanfox/docker-oldgit/blob/develop/Dockerfile)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    resource_class: xlarge
+    environment:
+      GRADLE_OPTS: -Dorg.gradle.console=plain -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false
```
Comment (iamdanfox) [inline .circleci/config.yml:12]: If you use the mega `com.palantir.baseline`, you get `com.palantir.baseline-circleci` which will detect these:

    CIRCLE_TEST_REPORTS: /home/circleci/junit
    CIRCLE_ARTIFACTS: /home/circleci/artifacts

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      - save_cache:
+          key: gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}
+          paths: [ ~/.gradle/caches ]
+      - run: ./gradlew --parallel --continue build
+      - run:
+          command: mkdir -p ~/junit && find . -type f -regex ".*/build/.*TEST.*xml" -exec cp --parents {} ~/junit/ \;
+          when: always
+      - store_test_results: { path: ~/junit }
+
+      - deploy:
```
Comment (iamdanfox) [inline .circleci/config.yml]: factor into a job so it won't run if `build-ibm` fails

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      - build:
+          filters:
+            # CircleCI2 will ignore tags without this. https://circleci.com/docs/2.0/workflows/#git-tag-job-execution
+            tags:
+              only: /.*/
+      - build-ibm:
+          filters:
+            tags:
+              only: /.*/
+
```
Comment (iamdanfox) [inline .circleci/config.yml]: `publish`

```yml
      - publish:
          requires: [ build, build-ibm ]
          filters: { tags: { only: /.*/ } }
```

File: .circleci/template.sh [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+export CIRCLECI_TEMPLATE=java-library-oss
```
Comment (iamdanfox) [inline .circleci/template.sh]: Delete this as it's already diverged?

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      - run:
+          command: mkdir -p ~/junit && find . -type f -regex ".*/build/.*TEST.*xml" -exec cp --parents {} ~/junit/ \;
+          when: always
+      - store_test_results: { path: ~/junit }
+
+      - deploy:
+          command: |
+            # publishing snapshots to bintray does not work, so we only publish from tag builds (not develop)
+            if [[ "${CIRCLE_TAG}" =~ [0-9]+(\.[0-9]+)+(-[a-zA-Z]+[0-9]*)* ]]; then
+              ./gradlew --stacktrace --continue publish
```
Comment (iamdanfox) [inline .circleci/config.yml]: gotta make sure `publish` will actually run `bintrayUpload`

(one option: https://github.com/palantir/conjure-java-runtime/blob/develop/gradle/publish-jar.gradle)

would need rootProject.name = 'hadoop-crypto' in your settings.gradle

Comment (ellisjoe) [reply .circleci/config.yml]: sweet. switched to `bintrayUpload` for now and will follow up with the rest

File: .circleci/config.yml (hunk 4 above)
Comment (ellisjoe) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 3 above)
Comment (ellisjoe) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 2 above)
Comment (ellisjoe) [reply .circleci/config.yml:12]: sweet. will follow up

File: .circleci/template.sh (hunk 5 above)
Comment (ellisjoe) [reply .circleci/template.sh]: done

File: .circleci/config.yml (hunk 1 above)
Comment (ellisjoe) [reply .circleci/config.yml]: done

**Review by iamdanfox — APPROVED** [review]

**Pull request merged**